packages/ses/index.d.ts

```diff
@@ -0,0 +1,112 @@
+type safetyPreference = 'safe' | 'unsafe';
+
+type LockdownOpts = {
+  fringeSet?: {
+    add?: Function,
+    has?: Function,
+  },
+  dateTaming?: safetyPreference,
+  errorTaming?: safetyPreference,
+  mathTaming?: safetyPreference,
+  regExpTaming?: safetyPreference,
+
+  [extraOption:string]?: any,
+};
+
+declare var lockdown = (LockdownOpts) => void;
+
+declare var harden = (any) => any;
+
+// Shared space by Realm
+type FullSpecifier = string;
+type ModuleNamespace = object;
+
+declare class StaticModuleRecord {
+  constructor (moduleSource: string, moduleLocation:string);
+  // intend to add reflection of import/export bindings
+
+  // needs to allow duplicates
+  staticImports(): {specifier: string, exportNames}[];
+}
+
+type CompartmentConstructorOptions = {
+  // JF has a better way for:
+  //   randomHook(): number; // Use for Math.random()
+  //   nowHook(): number; // Use for both Date.now() and new Date(),
+  // Supplied during Module instance creation instead of option
+  //   hasSourceTextAvailableHook(scriptOrModule): bool; // Used for censorship
+  resolveHook(name: string, referrer: FullSpecifier): FullSpecifier
+
+  // timing
+  // importHook delegates to importNowHook FIRST,
+  // they share a memo cache
+  // order to ensures importNow never allows async value of import
+  // to be accessed prior to any attached promise
+  importHook(fullSpec: FullSpecifier): Promise<StaticModuleRecord>;
+  importNowHook(fullSpec: FullSpecifier): StaticModuleRecord?;
```

Comment:
`importNowHook` is not implemented and might be unnecessary for the spec. I’ll definitely visit that when I spend some more time with the spec. It might be unnecessary since anything that the `importNowHook` could return could equally well be threaded through the compartment constructor’s module map.